### PR TITLE
HDDS-7135. ofs file input stream should support StreamCapabilities interface

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/HDDS-7135

Make sure that OFS returns an input stream that is wrapped by CapableOzoneFSInputStream when it is shaded with Hadoop3 dependency. CapableOzoneFSInputStream support StreamCapabilities API.

The StreamCapabilities API is added only in Hadoop3, so the Hadoop2-based filesystem jar returns the input stream wrapped by OzoneFSInputStream.

## How was this patch tested?
A simple assertion is added to validate the StreamCapabilities in the file system contrast test.
(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/contract/AbstractContractUnbufferTest.java [hunk 1]
```diff
@@ -54,6 +54,7 @@ public void setup() throws Exception {
   public void testUnbufferAfterRead() throws IOException {
     describe("unbuffer a file after a single read");
     try (FSDataInputStream stream = getFileSystem().open(file)) {
+      assertTrue(stream.hasCapability("in:unbuffer"));
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/contract/AbstractContractUnbufferTest.java:57]: `AbstractContractUnbufferTest` was copied from Hadoop when Ozone was still on 3.2.  I forgot to remove it  when upgrading to Hadoop 3.3.  This patch reminds me to do so.

Therefore I think we should add this assertion in `hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java` instead.

If you don't feel like updating the patch, I can make the move when removing this test class.

**Review by adoroszlai — COMMENTED** [review]
Thanks @jojochuang for the patch, LGTM.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/contract/AbstractContractUnbufferTest.java (hunk 1 above)
Comment (jojochuang) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/contract/AbstractContractUnbufferTest.java:57]: ok good to know. i'll update the PR soon.

**Review by adoroszlai — APPROVED** [review]
Thanks @jojochuang for updating the patch.

**Comment by smengcl** [comment]
Thanks @jojochuang for the patch, and @adoroszlai for reviewing and merging this.

Do we want to add a line in the doc to clarify that only Hadoop 3 supports streaming? I'm also assuming o3fs already supports `StreamCapabilities` (haven't checked myself)?

**Comment by adoroszlai** [comment]
> assuming o3fs already supports `StreamCapabilities`

Yes, it was added in HDDS-2750.

**Pull request merged**